### PR TITLE
Add authentication view with data validations

### PR DESCRIPTION
**Pull request by blacksam07**
## Summary
- Add authentication view (login and register) with data validations

## Trello Card
[8-register-layout](https://trello.com/c/yTbuElBU/8-register-layout)

## Screenshots

image

authentication-min

### CODE REVIEW

File: src/app/app.component.html [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <div>
```
Comment (CrisVega08) [inline src/app/app.component.html]: Why empty div?

File: src/app/app.component.ts [hunk 2]
```diff
@@ -7,4 +7,5 @@ import { Component } from '@angular/core';
 })
 export class AppComponent {
   title = 'wolox-books';
+  isLogin = 'true';
```
Comment (CrisVega08) [inline src/app/app.component.ts]: Why use a string?, try to use boolean

File: src/app/app.component.html [hunk 3]
```diff
@@ -1,5 +1,6 @@
...
-  <h1>wolox-books</h1>
+  <app-authentication >
+  </app-authentication>
 </div>
 
 <router-outlet></router-outlet>
```
Comment (CrisVega08) [inline src/app/app.component.html]: You don't are using routes so delete it.

File: src/app/authentication/authentication.component.html [hunk 4]
```diff
@@ -1 +1,27 @@
-<ng-content></ng-content>
```
Comment (CrisVega08) [inline src/app/authentication/authentication.component.html:1]: Why do you delete ??
I remember that card says you use a ng-content for showing maquetado

File: src/app/authentication/authentication.component.scss [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  border: 2px solid $earls-green;
+  border-radius: 5px;
+  box-sizing: border-box;
+  color: $earls-green;
+  cursor: pointer;
+  height: 35px;
+  margin: 20px 0;
+  text-align: center;
+  width: -webkit-fill-available;;
+}
```
Comment (CrisVega08) [inline src/app/authentication/authentication.component.scss]: Add empty line

File: src/app/authentication/authentication.component.ts [hunk 6]
```diff
@@ -1,15 +1,74 @@
...
+        validator: this.mustMatch('password', 'password_confirmation')
+      })
+    });
+  }
 
   ngOnInit(): void {
   }
 
+  onSubmit(value) {
+    value.user.locale = 'en'
```
Comment (CrisVega08) [inline src/app/authentication/authentication.component.ts]: you don't need this line, when you create form, you can add a default value for locate

File: src/app/authentication/authentication.component.html [hunk 7]
```diff
@@ -1 +1,27 @@
...
+<div class="login-container">
+  <form [formGroup]="authForm" (ngSubmit)="onSubmit(authForm.value)" [ngClass]="{'login-form': true, 'login-form-sign-up': !isLogin}" >
```
Comment (CrisVega08) [inline src/app/authentication/authentication.component.html]: when use reactive form don't need pass value on function 
```suggestion
  <form [formGroup]="authForm" (ngSubmit)="onSubmit()" [ngClass]="{'login-form': true, 'login-form-sign-up': !isLogin}" >
```

File: src/app/authentication/authentication.component.ts [hunk 8]
```diff
@@ -1,15 +1,74 @@
...
 
-  constructor() { }
+  constructor(private formBuilder: FormBuilder) {
+    this.authForm = this.formBuilder.group({
+      user: this.formBuilder.group({
+        first_name: '',
+        last_name: '',
+        email: [null, Validators.compose([Validators.required, Validators.email])],
+        password: [null, Validators.compose([Validators.required, Validators.pattern('[a-zA-Z]+[0-9]+')])],
+        password_confirmation: [''],
```
Comment (CrisVega08) [inline src/app/authentication/authentication.component.ts]: put validation required

File: src/app/authentication/authentication.component.html [hunk 9]
```diff
@@ -1 +1,27 @@
...
+    <div formGroupName="user">
+      <img class="form-logo" src="../assets/logo-wolox.png" alt="logo-wolox"/>
+      <div *ngIf="isLogin else signUp">
```
Comment (CrisVega08) [inline src/app/authentication/authentication.component.html]: separate this logic on two components one sign-up and other sign-in and use ng content-show the view

File: src/app/authentication/authentication.component.ts [hunk 10]
```diff
@@ -1,15 +1,74 @@
...
+    console.log(value);
+    this.authForm.reset();
+  }
+
+  checkInput(input: string) {
+    return !this.authForm.get('user')['controls'][input].valid && this.authForm.get('user')['controls'][input].touched;
+  }
+
+  checkPasswords(group: FormGroup) {
+  let pass = group.get('password').value;
```
Comment (CrisVega08) [inline src/app/authentication/authentication.component.ts]: Use const and check indentation

File: src/app/authentication/authentication.component.ts [hunk 11]
```diff
@@ -1,15 +1,74 @@
...
   ngOnInit(): void {
   }
 
+  onSubmit(value) {
+    value.user.locale = 'en'
+    console.log(value);
+    this.authForm.reset();
+  }
+
+  checkInput(input: string) {
```
Comment (CrisVega08) [inline src/app/authentication/authentication.component.ts]: Great :clap

File: src/app/authentication/authentication.component.ts [hunk 12]
```diff
@@ -1,15 +1,74 @@
...
 })
 export class AuthenticationComponent implements OnInit {
+  isLogin = true;
+  authForm: FormGroup;
+  label_button_up = 'Login';
+  label_button_down = 'Sign Up';
+  badEmail = false;
 
-  constructor() { }
+  constructor(private formBuilder: FormBuilder) {
```
Comment (CrisVega08) [inline src/app/authentication/authentication.component.ts]: Don't use constructor form, try to build form on onInit cycle

File: src/app/authentication/authentication.component.ts [hunk 13]
```diff
@@ -1,15 +1,74 @@
...
+    value.user.locale = 'en'
+    console.log(value);
+    this.authForm.reset();
+  }
+
+  checkInput(input: string) {
+    return !this.authForm.get('user')['controls'][input].valid && this.authForm.get('user')['controls'][input].touched;
+  }
+
+  checkPasswords(group: FormGroup) {
```
Comment (CrisVega08) [inline src/app/authentication/authentication.component.ts]:
```suggestion
  checkPasswords(group: : AbstractControl) {
```
Where do you use this function??

File: src/app/authentication/authentication.component.ts [hunk 14]
```diff
@@ -1,15 +1,74 @@
...
 
 @Component({
   selector: 'app-authentication',
   templateUrl: './authentication.component.html',
   styleUrls: ['./authentication.component.scss']
 })
 export class AuthenticationComponent implements OnInit {
+  isLogin = true;
+  authForm: FormGroup;
+  label_button_up = 'Login';
```
Comment (CrisVega08) [inline src/app/authentication/authentication.component.ts]: Use camelCase

File: src/app/authentication/authentication.component.ts [hunk 15]
```diff
@@ -1,15 +1,74 @@
...
+  label_button_down = 'Sign Up';
```
Comment (CrisVega08) [inline src/app/authentication/authentication.component.ts]: Same

File: src/app/authentication/authentication.component.ts [hunk 16]
```diff
@@ -1,15 +1,74 @@
...
+  checkPasswords(group: FormGroup) {
+  let pass = group.get('password').value;
+  let confirmPass = group.get('password_confirmation').value;
+
+  return pass === confirmPass ? null : { notSame: true }     
+  }
+
+  onSignUp () {
+    this.isLogin = !this.isLogin
+    if (!this.isLogin) this.authForm.get('user').get('password_confirmation').setValidators([Validators.required])
```
Comment (CrisVega08) [inline src/app/authentication/authentication.component.ts]: make this on onInit after change components to use ng-content 
```suggestion
    if (!this.isLogin) this.authForm.get('user.password_confirmation').setValidators([Validators.required])
```

File: src/app/authentication/authentication.component.ts [hunk 17]
```diff
@@ -1,15 +1,74 @@
...
+    let intermediate = this.label_button_up;
+    this.label_button_up = this.label_button_down;
+    this.label_button_down = intermediate;
+    this.authForm.reset();
+  }
+
+  mustMatch(controlName: string, matchingControlName: string) {
+    return (formGroup: FormGroup) => {
+        const control = formGroup.controls[controlName];
```
Comment (CrisVega08) [inline src/app/authentication/authentication.component.ts]: check indentation

File: src/app/authentication/authentication.component.ts [hunk 18]
```diff
@@ -1,15 +1,74 @@
...
+  onSignUp () {
+    this.isLogin = !this.isLogin
+    if (!this.isLogin) this.authForm.get('user').get('password_confirmation').setValidators([Validators.required])
+    let intermediate = this.label_button_up;
+    this.label_button_up = this.label_button_down;
+    this.label_button_down = intermediate;
+    this.authForm.reset();
+  }
+
+  mustMatch(controlName: string, matchingControlName: string) {
```
Comment (CrisVega08) [inline src/app/authentication/authentication.component.ts]: I dont understand this function I prefer that you use checksPasswords

**Review by CrisVega08 — CHANGES_REQUESTED** [review]

File: src/app/authentication/authentication.component.html [hunk 19]
```diff
@@ -1 +1,27 @@
...
+        <label class="label-input" for="apellido">Apellido</label>
+        <input class="form-input" id="apellido" type="text" formControlName="last_name">
+        <label class="label-input" for="email">Email</label>
+        <input class="form-input" [ngClass]="{'form-input-error': checkInput('email')}" id="email" type="text" formControlName="email">
+        <label class="label-input" for="password">Password</label>
+        <input class="form-input" [ngClass]="{'form-input-error': checkInput('password')}" id="password" type="password" formControlName="password">
+        <label class="label-input" for="password_confirmation">Confirmación de Password</label>
+        <input class="form-input" [ngClass]="{'form-input-error': checkInput('password_confirmation')}" id="password_confirmation" type="password" formControlName="password_confirmation">
+      </ng-template>
+      <button class="login-button" type="submit" [disabled]="!authForm.valid">{{label_button_up}}</button>
```
Comment (betoobernal) [inline src/app/authentication/authentication.component.html]: **only as note instructive**: you can to use `authForm.invalid`

File: src/app/authentication/authentication.component.ts [hunk 20]
```diff
@@ -1,15 +1,74 @@
...
+
+  checkInput(input: string) {
+    return !this.authForm.get('user')['controls'][input].valid && this.authForm.get('user')['controls'][input].touched;
+  }
+
+  checkPasswords(group: FormGroup) {
+  let pass = group.get('password').value;
+  let confirmPass = group.get('password_confirmation').value;
+
+  return pass === confirmPass ? null : { notSame: true }
```
Comment (betoobernal) [inline src/app/authentication/authentication.component.ts]: you can use an interruptor `pass === confirmPass && { notSame: true }`, unless that you will actually need to return a `null` value.

File: src/app/authentication/authentication.component.ts (hunk 16 above)
Comment (betoobernal) [inline src/app/authentication/authentication.component.ts]:
```suggestion
    if (!this.isLogin) {
        this.authForm.get('user')
            .get('password_confirmation')
            .setValidators([Validators.required])
    }
```

close it better in a block

File: src/app/authentication/authentication.component.html [hunk 21]
```diff
@@ -1 +1,27 @@
...
+        <label class="label-input" for="email">Email</label>
+        <input class="form-input" [ngClass]="{'form-input-error': checkInput('email')}" id="email" type="text" formControlName="email">
+        <label class="label-input" for="password">Password</label>
+        <input class="form-input" id="password" type="password" formControlName="password">
+      </div>
+      <ng-template #signUp>
+        <label class="label-input" for="nombre">Nombre</label>
+        <input class="form-input" id="nombre" type="text" formControlName="first_name">
+        <label class="label-input" for="apellido">Apellido</label>
+        <input class="form-input" id="apellido" type="text" formControlName="last_name">
```
Comment (betoobernal) [inline src/app/authentication/authentication.component.html:15]: what use do you give to `id` attribute?

File: src/app/app.component.ts (hunk 2 above)
Comment (blacksam07) [reply src/app/app.component.ts]: this code will be deleted in the future

File: src/app/app.component.ts [hunk 22]
```diff
@@ -1,9 +1,10 @@
...
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
-  styleUrls: ['./app.component.scss']
+  styleUrls: ['./app.component.scss'],
+  encapsulation: ViewEncapsulation.None
 })
 export class AppComponent {
   title = 'wolox-books';
```
Comment (CrisVega08) [inline src/app/app.component.ts]: Delete this property you didn't use

File: src/app/app.component.ts [hunk 23]
```diff
@@ -1,9 +1,10 @@
-import { Component } from '@angular/core';
+import { Component, ViewEncapsulation } from '@angular/core';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
-  styleUrls: ['./app.component.scss']
+  styleUrls: ['./app.component.scss'],
+  encapsulation: ViewEncapsulation.None
```
Comment (CrisVega08) [inline src/app/app.component.ts]: I  think you don't viewEncapsulation

File: src/app/authentication/authentication.component.scss [hunk 24]
```diff
@@ -0,0 +1,32 @@
+@import '../../styles.scss';
```
Comment (CrisVega08) [inline src/app/authentication/authentication.component.scss]: Delete you only need and try to use absolute paths
```suggestion
@import '../../_colors.scss';
```

File: src/app/authentication/authentication.component.scss [hunk 25]
```diff
@@ -0,0 +1,32 @@
...
+  width: 210px;
+}
+
+.signup-button {
+  &::after {
+    background-color: $silver-chalice;
+    content: '';
+    height: 2px;
+    left: 0;
+    position: absolute;
```
Comment (CrisVega08) [inline src/app/authentication/authentication.component.scss]: which is the reference for this position?

File: src/app/button/button.component.scss [hunk 26]
```diff
@@ -0,0 +1,38 @@
+@import '../../styles.scss';
```
Comment (CrisVega08) [inline src/app/button/button.component.scss]: Same

File: src/app/button/button.component.scss [hunk 27]
```diff
@@ -0,0 +1,38 @@
...
+  width: -webkit-fill-available;;
+}
+
+.border-line {
+  &::after {
+    background-color: $silver-chalice;
+    content: '';
+    height: 2px;
+    left: 0;
+    position: absolute;
```
Comment (CrisVega08) [inline src/app/button/button.component.scss:35]: Same about reference

File: src/app/button/button.component.ts [hunk 28]
```diff
@@ -0,0 +1,35 @@
+import { Component, EventEmitter, OnInit, Input, Output } from '@angular/core';
+
+@Component({
+  selector: 'app-button',
+  templateUrl: './button.component.html',
+  styleUrls: ['./button.component.scss']
+})
+export class ButtonComponent implements OnInit {
+  @Input('class')
+  bClass: string = 'primary';
```
Comment (CrisVega08) [inline src/app/button/button.component.ts]: one line

File: src/app/button/button.component.ts [hunk 29]
```diff
@@ -0,0 +1,35 @@
...
+
+  @Input('type')
```
Comment (CrisVega08) [inline src/app/button/button.component.ts]: same

File: src/app/signup/signup.component.html [hunk 30]
```diff
@@ -0,0 +1,18 @@
+<app-authentication>
+  <form [formGroup]="authForm">
+    <div formGroupName="user">
```
Comment (CrisVega08) [inline src/app/signup/signup.component.html:3]: Why you don't use (ngSubmit)="onSubmit()"

File: src/app/signup/signup.component.html [hunk 31]
```diff
@@ -0,0 +1,18 @@
...
+      <label class="label-input" for="email">Email</label>
+      <input class="form-input" [ngClass]="{'form-input-error': checkInput('email')}" type="text" formControlName="email">
+      <label class="label-input" for="password">Password</label>
+      <input class="form-input" [ngClass]="{'form-input-error': checkInput('password')}" type="password" formControlName="password">
+      <label class="label-input" for="password_confirmation">Confirmación de Password</label>
+      <input class="form-input" [ngClass]="{'form-input-error': checkInput('password_confirmation')}" type="password" formControlName="password_confirmation">
+    </div>
+  </form>
+  <app-button (bClick)="onSubmit(authForm.value)" [value]="'Sign Up'" [disabled]="authForm.invalid" [borderLine]="true"></app-button>
+  <app-button [class]="'secondary'" [value]="'Login'"></app-button>
```
Comment (CrisVega08) [inline src/app/signup/signup.component.html]: Dont do binding when is a constant 
```suggestion
  <app-button class='secondary' value="Login'></app-button>
```

File: src/app/signup/signup.component.html [hunk 32]
```diff
@@ -0,0 +1,18 @@
...
+      <input class="form-input" type="text" formControlName="last_name">
+      <label class="label-input" for="email">Email</label>
+      <input class="form-input" [ngClass]="{'form-input-error': checkInput('email')}" type="text" formControlName="email">
+      <label class="label-input" for="password">Password</label>
+      <input class="form-input" [ngClass]="{'form-input-error': checkInput('password')}" type="password" formControlName="password">
+      <label class="label-input" for="password_confirmation">Confirmación de Password</label>
+      <input class="form-input" [ngClass]="{'form-input-error': checkInput('password_confirmation')}" type="password" formControlName="password_confirmation">
+    </div>
+  </form>
+  <app-button (bClick)="onSubmit(authForm.value)" [value]="'Sign Up'" [disabled]="authForm.invalid" [borderLine]="true"></app-button>
```
Comment (CrisVega08) [inline src/app/signup/signup.component.html]:
```suggestion
  <app-button type="submit" value="Sign Up" [disabled]="authForm.invalid" [borderLine]="true"></app-button>
```

File: src/app/signup/signup.component.scss [hunk 33]
```diff
@@ -0,0 +1,43 @@
+@import '../../styles.scss';
```
Comment (CrisVega08) [inline src/app/signup/signup.component.scss]: Same here

File: src/app/signup/signup.component.ts [hunk 34]
```diff
@@ -0,0 +1,53 @@
...
+  templateUrl: './signup.component.html',
+  styleUrls: ['./signup.component.scss']
+})
+export class SignupComponent implements OnInit {
+  authForm: FormGroup;
+
+  constructor(private formBuilder: FormBuilder) {
+    this.authForm = this.formBuilder.group({
+      user: this.formBuilder.group({
+        first_name: '',
```
Comment (CrisVega08) [inline src/app/signup/signup.component.ts]: use camelCase always

File: src/app/signup/signup.component.ts [hunk 35]
```diff
@@ -0,0 +1,53 @@
...
+        last_name: '',
```
Comment (CrisVega08) [inline src/app/signup/signup.component.ts]: Same camelCase

File: src/app/signup/signup.component.ts [hunk 36]
```diff
@@ -0,0 +1,53 @@
...
+        email: [null, Validators.compose([Validators.required, Validators.email])],
+        password: [null, Validators.compose([Validators.required, Validators.pattern('[a-zA-Z]+[0-9]+')])],
+        password_confirmation: [''],
```
Comment (CrisVega08) [inline src/app/signup/signup.component.ts]: same

File: src/app/button/button.component.ts [hunk 37]
```diff
@@ -0,0 +1,35 @@
...
+  bType: string = 'button'
+
+  @Input('disabled')
+  bDisabled: boolean = false;
+
+  @Input('value')
```
Comment (CrisVega08) [inline src/app/button/button.component.ts]: very bad name jajaj i suggest label or text

File: src/app/button/button.component.ts [hunk 38]
```diff
@@ -0,0 +1,35 @@
...
+  value: string = 'Button';
+
+  @Input('borderLine')
```
Comment (CrisVega08) [inline src/app/button/button.component.ts]: one line

File: src/app/button/button.component.ts [hunk 39]
```diff
@@ -0,0 +1,35 @@
...
+  borderLine: boolean = false;
+
+  @Output() bClick = new EventEmitter();
+
+  constructor() { }
+
+  ngOnInit(): void {
+    if (this.borderLine) this.bClass = `${this.bClass} border-line`
```
Comment (CrisVega08) [inline src/app/button/button.component.ts]: Do this in template

**Review by CrisVega08 — CHANGES_REQUESTED** [review]

File: src/app/authentication/authentication.component.html [hunk 40]
```diff
@@ -1 +1,6 @@
-<ng-content></ng-content>
+<div class="authentication-container">
+  <div class="authentication-form" >
```
Comment (CrisVega08) [inline src/app/authentication/authentication.component.html:2]: problems with indentation

File: src/app/authentication/authentication.component.scss [hunk 41]
```diff
@@ -0,0 +1,20 @@
+@import '_colors';
```
Comment (CrisVega08) [inline src/app/authentication/authentication.component.scss:1]: Nice

File: src/app/button/button.component.ts [hunk 42]
```diff
@@ -0,0 +1,33 @@
+import { Component, EventEmitter, OnInit, Input, Output } from '@angular/core';
+
+@Component({
+  selector: 'app-button',
+  templateUrl: './button.component.html',
+  styleUrls: ['./button.component.scss']
+})
+export class ButtonComponent implements OnInit {
+  @Input('class')
```
Comment (CrisVega08) [inline src/app/button/button.component.ts]: I forget say you, don't make aliases, change on names o inputs for example 

@Input() class = 'primary';

File: src/app/signup/signup.component.html [hunk 43]
```diff
@@ -0,0 +1,18 @@
+<app-authentication>
+  <form [formGroup]="authForm" (ngSubmit)="onSubmit(authForm.value)">
```
Comment (CrisVega08) [inline src/app/signup/signup.component.html]: Dont put value, obtain it on component with this.authForm.value
```suggestion
  <form [formGroup]="authForm" (ngSubmit)="onSubmit()">
```

File: src/app/signup/signup.component.html [hunk 44]
```diff
@@ -0,0 +1,18 @@
...
+      <label class="label-input" for="apellido">Apellido</label>
+      <input class="form-input" type="text" formControlName="lastName">
+      <label class="label-input" for="email">Email</label>
+      <input class="form-input" [ngClass]="{'form-input-error': checkInput('email')}" type="text" formControlName="email">
+      <label class="label-input" for="password">Password</label>
+      <input class="form-input" [ngClass]="{'form-input-error': checkInput('password')}" type="password" formControlName="password">
+      <label class="label-input" for="password_confirmation">Confirmación de Password</label>
+      <input class="form-input" [ngClass]="{'form-input-error': checkInput('passwordConfirmation')}" type="password" formControlName="passwordConfirmation">
+    </div>
+    <app-button [type]="'submit'" [label]="'Sign Up'" [disabled]="authForm.invalid" [oneLine]="true"></app-button>
```
Comment (CrisVega08) [inline src/app/signup/signup.component.html]: Remember when value is a string don't do binding
```suggestion
    <app-button type="submit" label="Sign Up" [disabled]="authForm.invalid" [oneLine]="true"></app-button>
```

File: src/app/signup/signup.component.html [hunk 45]
```diff
@@ -0,0 +1,18 @@
...
+    <app-button [class]="'secondary'" [label]="'Login'"></app-button>
```
Comment (CrisVega08) [inline src/app/signup/signup.component.html]: Same

File: src/app/signup/signup.component.ts [hunk 46]
```diff
@@ -0,0 +1,60 @@
...
+  templateUrl: './signup.component.html',
+  styleUrls: ['./signup.component.scss']
+})
+export class SignupComponent implements OnInit {
+  authForm: FormGroup;
+
+  constructor(private formBuilder: FormBuilder) {
+    this.authForm = this.formBuilder.group({
+      user: this.formBuilder.group({
+        firstName: '',
```
Comment (CrisVega08) [inline src/app/signup/signup.component.ts]: put required validator

File: src/app/signup/signup.component.ts [hunk 47]
```diff
@@ -0,0 +1,60 @@
...
+        lastName: '',
```
Comment (CrisVega08) [inline src/app/signup/signup.component.ts]: Same

File: src/app/signup/signup.component.ts [hunk 48]
```diff
@@ -0,0 +1,60 @@
...
+        validator: this.checkConfirmation('password', 'passwordConfirmation')
+      })
+    });
+  }
+
+  ngOnInit(): void {
+  }
+
+  onSubmit(value) {
+    value.user.locale = 'en'
```
Comment (CrisVega08) [inline src/app/signup/signup.component.ts]: You don't need it, you are setting at creating form

**Review by CrisVega08 — CHANGES_REQUESTED** [review]

File: src/app/signup/signup.component.ts [hunk 49]
```diff
@@ -0,0 +1,59 @@
...
+  toSnakeCase = value => {
+    return value.replace(/([A-Z])/g, letter => `_${letter.toLowerCase()}`);
+  }
+
+  renameKeys (obj) {
+    let res: Object = {};
+    Object.keys(obj).map(key => res[this.toSnakeCase(key)] = obj[key]);
+    return res;
+  }
+
```
Comment (betoobernal) [inline src/app/signup/signup.component.ts]: these methods could are in a utils file 🤔

File: src/app/signup/signup.component.ts [hunk 50]
```diff
@@ -0,0 +1,59 @@
...
+@Component({
+  selector: 'app-signup',
+  templateUrl: './signup.component.html',
+  styleUrls: ['./signup.component.scss']
+})
+export class SignupComponent implements OnInit {
+  authForm: FormGroup;
+
+  constructor(private formBuilder: FormBuilder) {
+    this.authForm = this.formBuilder.group({
```
Comment (CrisVega08) [inline src/app/signup/signup.component.ts:16]: other thing remove this on constructor() and put inside onInit().

**Review by CrisVega08 — CHANGES_REQUESTED** [review]

File: src/app/helpers/utils/utils.ts [hunk 51]
```diff
@@ -0,0 +1,9 @@
+export function toSnakeCase (value: string) {
+  return value.replace(/([A-Z])/g, letter => `_${letter.toLowerCase()}`);
+};
+
+export function keysToSnakeCase (obj: Object) {
+  let res: Object = {};
```
Comment (betoobernal) [inline src/app/helpers/utils/utils.ts]: the res variable doesn't need a typing, she gets one implicit when it's assigned.

**Review by CrisVega08 — APPROVED** [review]

**Review by betoobernal — APPROVED** [review]

**Pull request merged**